cli/api/daemon.go

```diff
@@ -671,7 +671,8 @@ func (d *daemon) startAgent() error {
 	// TODO: Integrate this server into the rpc server, or something.
 	// Currently its only use is for command execution.
 	go func() {
-		sio := shell.NewProcessExecutorServer(options.Endpoint, options.DockerRegistry, options.ControllerBinary, options.UIPort)
+		agentEndpoint := "localhost:" + options.RPCPort
```

Comment:
I'm confused about when we are supposed to use localhost vs 127.0.0.1, but looks like localhost is used more than 127.0.0.1 in our code.